daffodil-runtime2/src/main/resources/c/libruntime/infoset.h

```diff
@@ -102,14 +100,16 @@ typedef struct InfosetBase
 
 typedef struct PState
 {
-    FILE *stream; // input to read from
+    FILE *      stream;    // input to read from
+    const char *error_msg; // to stop if an error happens
```

Comment:
I'm not sure when we will need to support backtracking in a binary representation with only a limited subset of DFDL 1.0 supported.  A collaborator my group is working with has been using `dfdl:discriminator test="{../../tagt eq 111}"` on message elements to perform discrimination from a choice of equally valid message elements, but I think his schema should be using dfdl:choiceDispatchKey instead which doesn't require backtracking.  When and where might we start to need backtracking, I'm wondering?

Reply:
Not soon. My preference is create the deterministic DFDL subset first. Eventually people will want a fully functioning back-end.

Reply:
The other reason to do this is so that programmers won't write code assembling error messages from strings. They'll just pass a bunch of static pointers. This is less code clutter.